phac-nml/irida-next
# GraphQL: Add projectSample query [DFCT0010065]

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in a new query named `projectSample` which returns a `Sample` if found within a `Project`. The query find the `Sample` by either its puid or its name.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Launch server and login as your preferred user
2. Navigate to `GraphiQL` endpoint (e.g. `http://localhost:3000/graphiql`)
3. Try the `projectSample` query using sample puid
```graphql
{
  projectSample(projectPuid: "PROJECT_PUID", samplePuid: "SAMPLE_PUID") {
    name
    puid
  }
}
```
4. Try the `projectSample` query using sample name
```graphql
{
  projectSample(projectPuid: "PROJECT_PUID", sampleName: "SAMPLE NAME") {
    name
    puid
  }
}
```
5. Try querying for a Sample which belongs to a different project and ensure no result

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
